crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1001 @@
+        let mut ambiguous_with_flattened = UnGraphMap::new();
+        for (lhs, rhs, _) in self.ambiguous_with.all_edges() {
+            match (lhs, rhs) {
+                (NodeId::System(_), NodeId::System(_)) => {
+                    ambiguous_with_flattened.add_edge(lhs, rhs, ());
+                }
+                (NodeId::Set(_), NodeId::System(_)) => {
+                    for &lhs_ in systems[&lhs].iter() {
+                        ambiguous_with_flattened.add_edge(lhs_, rhs, ());
+                    }
+                }
+                (NodeId::System(_), NodeId::Set(_)) => {
+                    for &rhs_ in systems[&rhs].iter() {
+                        ambiguous_with_flattened.add_edge(lhs, rhs_, ());
+                    }
+                }
+                (NodeId::Set(_), NodeId::Set(_)) => {
+                    for &lhs_ in systems[&lhs].iter() {
+                        for &rhs_ in systems[&rhs].iter() {
+                            ambiguous_with_flattened.add_edge(lhs_, rhs_, ());
+                        }
+                    }
+                }
+            }
+        }
+
+        self.ambiguous_with_flattened = ambiguous_with_flattened;
+
+        // check for conflicts
+        let mut conflicting_systems = Vec::new();
+        for &(a, b) in flat_results.disconnected.iter() {
+            if self.ambiguous_with_flattened.contains_edge(a, b)
+                || self.ambiguous_with_all.contains(&a)
+                || self.ambiguous_with_all.contains(&b)
+            {
+                continue;
+            }
+
+            let system_a = self.systems.get(&a).unwrap();
+            let system_b = self.systems.get(&b).unwrap();
+            if system_a.is_exclusive() || system_b.is_exclusive() {
+                conflicting_systems.push((a, b, Vec::new()));
+            } else {
+                let access_a = system_a.component_access();
+                let access_b = system_b.component_access();
+                if !access_a.is_compatible(&access_b) {
+                    let conflicts = access_a.get_conflicts(access_b);
+                    conflicting_systems.push((a, b, conflicts));
+                }
+            }
+        }
+
+        if self.contains_conflicts(&conflicting_systems) {
+            self.report_conflicts(&conflicting_systems);
+            return Err(BuildError::Ambiguity);
+        }
+
+        Ok(())
+    }
+
+    fn build_schedule(&mut self) -> SystemSchedule {
+        let sys_count = self.systems.len();
+        let node_count = self.systems.len() + self.system_sets.len();
+
+        let dg_system_ids = self.dependency_flattened.topsort.clone();
+        let dg_system_idx_map = dg_system_ids
+            .iter()
+            .cloned()
+            .enumerate()
+            .map(|(i, id)| (id, i))
+            .collect::<HashMap<_, _>>();
+
+        // get the number of dependencies and the immediate dependents of each system
+        // (needed to run systems in the correct order)
```

Comment:
```suggestion
        // (needed to run systems in the correct order as completed dependencies are subtracted one at a time)
```

Reply:
Tried to clarify that this specific metadata is for multi-threaded executor.